pkg/virt-controller/services/template.go

```diff
@@ -1456,6 +1457,8 @@ func (t *templateService) VMIResourcePredicates(vmi *v1.VirtualMachineInstance,
 		vmiCPUArch = t.clusterConfig.GetClusterCPUArch()
 	}
 	memoryOverhead := GetMemoryOverhead(vmi, vmiCPUArch, t.clusterConfig.GetConfig().AdditionalGuestMemoryOverheadRatio)
+	memoryOverhead.Add(netbinding.CalculateMemoryOverhead(vmi, t.clusterConfig.GetNetworkBindings()))
```

Comment:
@orelmisan the `GetMemoryOverhead` is a public function, why didn't we add the GetNetworkBindings overhead inside?

Reply:
@vladikr the new memory overhead calculation for the network binding plugins was not integrated into `GetMemoryOverhead` because the `GetMemoryOverhead` is invoked from multiple locations, some do not have access to cluster config which is required for the new calculation. Additionally, changing the signature of `GetMemoryOverhead` would have caused multiple tests to break (and would have made the PR much larger).
By calling the new calculation from the outside, we ensure that all necessary data is available to it, preserve the integrity of existing tests and decouple the logic.

Reply:
AFAIR the reason that convinced me it does not belong there was that the helper was called from multiple locations from which not all current correct and with the addition we needed, will surely not be correct.

One example is the call from the `virt-handler` [1], where this additional memory should not be considered. Some other memory overheads should not be considered there as well, but that is a story for a fix and not for an addition.

If it would have been related to just a signature change and modifying tests, I would have considered it a weak reasoning.

I guess an overall fix would be to separate the memory resource calculation for the spec and the the one for the runtime. That requires a dedicated task.

[1] https://github.com/orelmisan/kubevirt/blob/fc9301d5b70ef49cd0227b896b7c58fe54ff8a6b/pkg/virt-handler/isolation/detector.go#L140